R/verify-variables.R

```diff
@@ -6,26 +6,31 @@
 #' occurs, rather than within this function.
 #'
 #' @param data The dataset to check.
+#' @param call The environment where the function is called, so that the error
+#'   traceback gives a more meaningful location.
 #' @inheritParams get_required_variables
 #'
-#' @return Either TRUE if the verification passes, or a character string if
-#'   there is an error.
+#' @return Either TRUE if the verification passes, or an error.
 #' @keywords internal
 #'
 #' @examples
 #' # TODO: Replace with simulated data.
-#' example_bef_data <- tibble::tibble(pnr = 1, koen = 1, foed_dato = 1)
-#' osdc:::verify_required_variables(example_bef_data, "bef")
-verify_required_variables <- function(data, register) {
+#' verify_required_variables(register_data$bef, "bef")
+#' verify_required_variables(register_data$bef, "lpr_adm")
+verify_required_variables <- function(data, register, call = rlang::caller_env()) {
```

Comment:
The `call` argument and `rlang::caller_env()` allows the error message to occur in the function this is used in. That way, the log traceback message doesn't show this function, only the function where the error should be shown (e.g. `join_lpr2()`).

Reply:
That's neat!